### PR TITLE
Add clean.ps1 and clean.sh

### PR DESCRIPTION
**Pull request by dougbu**
- #10463

### CODE REVIEW

File: clean.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+        *)
+            git_args[${#git_args[*]}]="$1"
+            ;;
+    esac
+    shift
+done
+
+# This incantation avoids unbound variable issues if git_args is empty
+# https://stackoverflow.com/questions/7577052/bash-empty-array-expansion-with-set-u
+git clean -dix -e .dotnet/ -e .tools/ -e src/SignalR/clients/ts/FunctionalTests/node_modules/ ${git_args[@]+"${git_args[@]}"}
```
Comment (natemcmaster) [inline clean.sh]:
```suggestion
git clean -dix -e .dotnet/ -e .tools/ ${git_args[@]+"${git_args[@]}"}
```

Git on non-Windows handles symlinks correctly, so we don't need this exclusion :)

**Review by natemcmaster — APPROVED** [review]
Seems fine to me. Would be good to get @Tratcher's approval -- he was one of the devs requesting this.

**Comment by Tratcher** [comment]
That looks helpful. How about cmd? We have cmd for most other common commands.

**Comment by dougbu** [comment]
Took your suggestion @Tratcher

**Review by Tratcher — APPROVED** [review]

**Comment by dougbu** [comment]
Another now-standard timeout of the build. Merging since issue can't be related to these three new files.

**Pull request merged**